### PR TITLE
chore(markdown): replaces Markdown library with composer managed version

### LINKED ISSUE #8597: replace php markdown with composer alternative
I was just looking at stuff we could cleanup...

we only have 2 tools in /vendors left (markdown and colorbox-theme).

We could replace the markdown php library with a client side thing like https://github.com/showdownjs/showdown (or something else... i did not look for the 'best') solution

### PR DESCRIPTION
**Pull request by jeabakker**
The Markdown library is available as a composer package, this requires
less maintenance by Elgg developers.

fixes #8597
- [x] `BREAKING CHANGE:` notice in commit
- [x] mention in docs/guide/upgrading.rst that the `elgg:markdown` is removed

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -30,7 +30,8 @@
         "bower-asset/jquery-colorbox": "^1.5.14",
         "bower-asset/jquery-imgareaselect": "^0.9.0",
         "bower-asset/jquery-ui-extensions": "dev-master",
-        "bower-asset/jquery-jeditable": "^1.7.3"
+        "bower-asset/jquery-jeditable": "^1.7.3",
+        "michelf/php-markdown": "^1.5.0"
```
Comment (mrclay) [inline composer.json]: Should we put this above the bower stuff?

**Comment by mrclay** [comment]
Needs to include a `BREAKING CHANGE:` notice in commit and to mention in docs/guide/upgrading.rst that the `elgg:markdown` is removed.

File: composer.json (hunk 1 above)
Comment (jeabakker) [inline composer.json]: any reason? it's something new and thus i added it at the bottom

Comment (mrclay) [inline composer.json]: I figured we'd keep all the PHP libs together. Not important.

**Comment by jeabakker** [comment]
did my to-dos

File: docs/guides/upgrading.rst [hunk 2]
```diff
@@ -454,6 +454,11 @@ Also several workarounds for very old views are no longer performed. Make these
  - Set ``$vars['name']`` instead of ``$vars['internalname']``.
  - Set ``$vars['id']`` instead of ``$vars['internalid']``.
 
+Removed libraries
+-----------------
+
+ - ``elgg.markdown``: replace with ``use \Michelf\MarkdownExtra``
```
Comment (mrclay) [inline docs/guides/upgrading.rst]:
```
- ``elgg.markdown``: Elgg no longer provides a markdown implementation. You must provide your own.
```

Comment (jeabakker) [inline docs/guides/upgrading.rst]: not true, the same markdown implementation is still the same (newer version) only need a newer call

Comment (mrclay) [inline docs/guides/upgrading.rst]: We don't want to bind ourselves to providing this implementation until 3.0.

Comment (mrclay) [inline docs/guides/upgrading.rst]: @Elgg/owners is our intention to keep providing MarkdownExtra as part of our API?

Comment (jeabakker) [inline docs/guides/upgrading.rst]: > @Elgg/owners is our intention to keep providing MarkdownExtra as part of our API?

please keep in mind that this applies to all composer libs ;)

Comment (ewinslow) [inline docs/guides/upgrading.rst]: Agree with Steve: let's not encourage folks to have indirect dependencies. Removing one of our requirements should be considered a BC change.

Comment (jeabakker) [inline docs/guides/upgrading.rst]: will continue the discussion in #7080

**Comment by ewinslow** [comment]
Needs rebase. Also the library name should have a colon not a dot. Minor issue though.

**Comment by jeabakker** [comment]
rebased and fixed the dot to a colon ;)

**Pull request merged**